### PR TITLE
Manual tests support

### PR DESCRIPTION
**Pull request by mvdr**
---

> team(s) or team member(s) automatically mentioned by [CodeGurusBot](https://github.com/NIPOSoftwareBV/Nfield-Documentation/blob/master/Agreements/codegurus.md). If you want to review this PR, please self-request a review.

@NIPOSoftwareBV/team-purple

### CODE REVIEW

File: Library/Models/ManualTestSurveyResult.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    {
+        /// <summary>
+        /// Survey id of the original survey
+        /// </summary>
+        public string OriginalSurveyId { get; set; }
+
+        /// <summary>
+        /// Survey id of the manual test survey
+        /// </summary>
+        public string SimulationSurveyId { get; set; }
```
Comment (pilarodriguez) [inline Library/Models/ManualTestSurveyResult.cs:30]: This should be TestSurveyId or ManualTestSurveyId

File: Library/Models/StartCreateManualTestSurvey.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+//    You should have received a copy of the GNU Lesser General Public License
+//    along with Nfield.SDK.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace Nfield.SDK.Models
+{
+    /// <summary>
+    /// Parameters to start creating a manual test survey.
+    /// To specify uploaded files this model uses the file name and the file content.
+    /// For the sample data we expect binary as an input since it could be csv or Excel data. 
+    /// To upload data directly from files use <see cref="InterviewSimulationFiles"/>
```
Comment (pilarodriguez) [inline Library/Models/StartCreateManualTestSurvey.cs:22]: I think this should be StartCreateManualTestSurveyFile

File: Library/Services/INfieldSurveyManualTestsService.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        /// <summary>
+        /// Get manual tests of a specific survey (normally just one)
+        /// </summary>
+        Task<IQueryable<SurveyManualTest>> GetSurveyManualTestsAsync(string surveyId);
+
+        /// <summary>
+        /// Get manual test of a specific survey
+        /// </summary>
+        Task<SurveyManualTest> GetSurveyManualTestAsync(string surveyId, string manualTestId);
```
Comment (pilarodriguez) [inline Library/Services/INfieldSurveyManualTestsService.cs:36]: I don't understand these endpoints. What are they returning? I would expect that we have 1 endpoint without any parameters to return all the manual test surveys in the domain, and another endpoint that for a given survey id, you return the manual test survey

Comment (mvdr) [reply Library/Services/INfieldSurveyManualTestsService.cs:36]: The endpoint to return all manual test surveys in the domain is the GetAsync method in NfieldManualTestsService class. This class is for the one with the given surveyId => GetSurveyManualTestsAsync(string surveyId); It returns an array with normally just one test survey. Then we have another endpoint that returns just the one test survey => GetSurveyManualTestAsync(string surveyId, string manualTestId);    You could say that the last one is not needed. But Team Yellow preferred it that way. It is conform the REST standards.

Comment (pilarodriguez) [reply Library/Services/INfieldSurveyManualTestsService.cs:36]: But why we need an endpoint for this? If they want to get the manual test survey, they can use the regular GetSurvey endpoint using the manualTestId as surveyId

Comment (mvdr) [reply Library/Services/INfieldSurveyManualTestsService.cs:36]: The GetSurvey endpoint does not return the OriginalSurveyId though. But on the other hand, you would have to know the OriginalSurvey to use the new (maybe not needed) endpoint.

**Pull request merged**